flytekit/control_plane/identifier.py

```diff
@@ -0,0 +1,137 @@
+import six as _six
+
+from flytekit.common.exceptions import user as _user_exceptions
+from flytekit.models.core import identifier as _core_identifier
+
+
+class Identifier(_core_identifier.Identifier):
+
+    _STRING_TO_TYPE_MAP = {
+        "lp": _core_identifier.ResourceType.LAUNCH_PLAN,
+        "wf": _core_identifier.ResourceType.WORKFLOW,
+        "tsk": _core_identifier.ResourceType.TASK,
+    }
+    _TYPE_TO_STRING_MAP = {v: k for k, v in _six.iteritems(_STRING_TO_TYPE_MAP)}
+
+    @classmethod
+    def promote_from_model(cls, base_model: _core_identifier.Identifier) -> "Identifier":
+        return cls(base_model.response_type, base_model.project, base_model.domain, base_model.name, base_model.version)
+
+    @classmethod
+    def from_python_std(cls, string: str) -> "Identifier":
```

Comment:
It took me two months to figure out what "python std" meant... can we just call it from urn or from string or something?